ocs_ci/deployment/deployment.py

```diff
@@ -169,22 +178,149 @@ def deploy_cluster(self, log_cli_level="DEBUG"):
+                mode='w+', prefix="odf_multicluster_orchestrator", delete=False
+            )
+            templating.dump_data_to_temp_yaml(
+               odf_multicluster_orchestrator_data,
+               odf_multicluster_orchestrator.name 
+            )
+            run_cmd(f"oc create -f {odf_multicluster_orchestrator.name}")
+            self.configure_mirror_peer()
+
+    def configure_mirror_peer(self):
+        # Create mirror peer 
+        mirror_peer_data = templating.load_yaml(
+            constants.MIRROR_PEER
+        )
+        mirror_peer_yaml = tempfile.NamedTemporaryFile(
+            mode='w+', prefix="mirror_peer", delete=False 
+        )
+        # Update all the participating clusters in mirror_peer_yaml
+        non_acm_clusters = get_non_acm_cluster_config()
+        primary = get_primary_cluster_config()
+        non_acm_clusters.remove(primary)
+        index = -1 
+        # First entry should be the primary cluster
+        # in the mirror peer
+        for cluster_entry in mirror_peer_data['spec']['items']:
+            if index == -1:
+                cluster_entry["clusterName"] = primary["ENV_DATA"]["cluster_name"]
+            else:
+                cluster_entry["clusterName"] = (
+                    non_acm_clusters[index]["ENV_DATA"]["cluster_name"]
+                )
+                index =+ 1
+        templating.dump_data_to_temp_yaml(
+            mirror_peer_data,
+            mirror_peer_yaml.name
+        )
+        config.switch_acm_ctx()
+        run_cmd(f"oc create -f {mirror_peer_yaml.name}")
+        self.validate_mirror_peer()
+        # Patch storagecluster on all the DR participating  clusters(except ACM)
+        index = 0
+        for cluster in config.clusters:
+            if config.clusters.index(cluster) == config.acm_index:
+                continue
+            else:
+                config.switch_ctx(index)
+                patch_cmd = f"oc patch storagecluster \${constants.rbd_mirroring_storagecluster_patch}"
+
+                run_cmd(patch_cmd)
+                query_mirroring = (
+                    f"oc get CephBlockPool -n {constants.OPENSHIFT_STORAGE_NAMESPACE}"
+                )
+                out = run_cmd(query_mirroring)
+                if out != "true":
+                    logger.error(f"On cluster {cluster.ENV_DATA['cluster_name']}")
+                    ResourceWrongStatusException("CephBlockPool", expected='true', got=out)
+                index =+ 1
+        
+    def validate_mirror_peer(self):
+        """
+        Validate mirror peer
+
+        1. Check initial phase of 'ExchangingSecret'
+        2. Check token-exchange-agent pod in 'Running' phase
+
+        """
+        # Check mirror peer status only on HUB
+        mirror_peer = ocp.OCP(
+            kind='MirrorPeer', 
+            namespace=constants.dr_default_namespace
+        )
+        mirror_peer.wait_for_phase(phase="ExchangingSecret")
```

Comment:
we should have 2 condition some time we also see `ExchangedSecret`

Reply:
I have added both condition one after the other, will it be serial in changing the states OR we can expect either one of this ?